test/functional/insert.test.js

```diff
@@ -2583,4 +2584,45 @@ describe('Insert', function () {
       });
     }
   });
+
+  it('MongoBulkWriteError.insertedCount should respect BulkWriteResult.insertedCount should respect BulkWrite.nInserted', function () {
```

Comment:
```suggestion
  it('MongoBulkWriteError.insertedCount should respect BulkWrite.nInserted', function () {
```

Reply:
I'll give this a better name but this was actually correct `MongoBulkWriteError` holds a `BulkWriteResult` which holds the `BulkWrite` so this test makes sure the values thread correctly through the two wrapping classes